### ISSUE TITLE
iptables: update would break connmand

### ISSUE THREAD

**Issue by mdorman** [NixOS/nixpkgs#12177]
I recently did `nixos-rebuild switch --no-build-output --option binary-caches http://hydra.nixos.org/ -I nixpkgs=/home/mdorman/src/nixpkgs` (where `~mdorman/src/nixpkgs` is an up-to-date repo) and my networking broke.  The specific issue was `connmand` segfaulting:

```
Jan 05 22:02:06 aching connmand[7866]: Connection Manager version 1.30
Jan 05 22:02:06 aching audit: NETFILTER_CFG table=filter family=2 entries=27
Jan 05 22:02:06 aching audit[7866]: SYSCALL arch=c000003e syscall=54 success=yes exit=0 a0=a a1=0 a2=40 a3=103a5b0 items=0 ppid=1 pid=7866 auid=4294967295 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0 tty=(none) ses=4294967295 comm="connmand" exe="/nix/store/x7lyis4srvd68lflgnqqmg2bc1fm2whb-connman-1.30/bin/connmand" key=(null)
Jan 05 22:02:06 aching audit: PROCTITLE proctitle=2F6E69782F73746F72652F78376C796973347372766436386C666C676E71716D6732626331666D327768622D636F6E6E6D616E2D312E33302F7362696E2F636F6E6E6D616E64002D2D636F6E6669673D2F6E69782F73746F72652F37383078797137726367376766306271706A3130306C666238336B69367938762D636F6E6E
Jan 05 22:02:06 aching audit: NETFILTER_CFG table=mangle family=2 entries=6
Jan 05 22:02:06 aching audit[7866]: SYSCALL arch=c000003e syscall=54 success=yes exit=0 a0=a a1=0 a2=40 a3=1038c00 items=0 ppid=1 pid=7866 auid=4294967295 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0 tty=(none) ses=4294967295 comm="connmand" exe="/nix/store/x7lyis4srvd68lflgnqqmg2bc1fm2whb-connman-1.30/bin/connmand" key=(null)
Jan 05 22:02:06 aching audit: PROCTITLE proctitle=2F6E69782F73746F72652F78376C796973347372766436386C666C676E71716D6732626331666D327768622D636F6E6E6D616E2D312E33302F7362696E2F636F6E6E6D616E64002D2D636F6E6669673D2F6E69782F73746F72652F37383078797137726367376766306271706A3130306C666238336B69367938762D636F6E6E
Jan 05 22:02:06 aching audit: NETFILTER_CFG table=nat family=2 entries=5
Jan 05 22:02:06 aching audit[7866]: SYSCALL arch=c000003e syscall=54 success=yes exit=0 a0=a a1=0 a2=40 a3=1037800 items=0 ppid=1 pid=7866 auid=4294967295 uid=0 gid=0 euid=0 suid=0 fsuid=0 egid=0 sgid=0 fsgid=0 tty=(none) ses=4294967295 comm="connmand" exe="/nix/store/x7lyis4srvd68lflgnqqmg2bc1fm2whb-connman-1.30/bin/connmand" key=(null)
Jan 05 22:02:06 aching audit: PROCTITLE proctitle=2F6E69782F73746F72652F78376C796973347372766436386C666C676E71716D6732626331666D327768622D636F6E6E6D616E2D312E33302F7362696E2F636F6E6E6D616E64002D2D636F6E6669673D2F6E69782F73746F72652F37383078797137726367376766306271706A3130306C666238336B69367938762D636F6E6E
Jan 05 22:02:06 aching connmand[7866]: Aborting (signal 11) [/nix/store/x7lyis4srvd68lflgnqqmg2bc1fm2whb-connman-1.30/sbin/connmand]
Jan 05 22:02:06 aching connmand[7866]: ++++++++ backtrace ++++++++
Jan 05 22:02:06 aching connmand[7866]: +++++++++++++++++++++++++++
Jan 05 22:02:06 aching systemd[1]: connman.service: Main process exited, code=exited, status=1/FAILURE
Jan 05 22:02:06 aching systemd[1]: connman.service: Unit entered failed state.
Jan 05 22:02:06 aching systemd[1]: connman.service: Failed with result 'exit-code'.
Jan 05 22:02:06 aching systemd[1]: connman.service: Service hold-off time over, scheduling restart.
Jan 05 22:02:06 aching systemd[1]: Stopped Connection service.
```

I tried several things to resolve it, including upgrading my kernel to 4.3.3, upgrading connman to 1.31.  Neither resolved the issue, so I took the hash of the last version that I knew worked (74fc8dc0ced8368c7196e7778b2b1ab99adafad6) and did a bisection---judging strictly on whether my networking died---with the following results:

```
git bisect start
# bad: [74fc8dc0ced8368c7196e7778b2b1ab99adafad6] fix ding desktop file path
git bisect bad 74fc8dc0ced8368c7196e7778b2b1ab99adafad6
# good: [93d8671e2c6d1d25f126ed30e5e6f16764330119] nixos/rabbitmq: fix link to documentation
git bisect good 93d8671e2c6d1d25f126ed30e5e6f16764330119
# skip: [db736b57fda7e533fc7a6956995e5285748c63d4] Merge pull request #11806 from deeky666/php-7.0.1
git bisect skip db736b57fda7e533fc7a6956995e5285748c63d4
# skip: [4ba31045e46b54e8b3b3d240858f83a60b9ef49c] nixpkgs manual: more strict commit policy (close #11609)
git bisect skip 4ba31045e46b54e8b3b3d240858f83a60b9ef49c
# skip: [99816323f9c475d8dc67af85af85ba7269c15cf7] saneBackendsGit 1.0.25.{177-g753d123 -> 180-g6d8b8d5}
git bisect skip 99816323f9c475d8dc67af85af85ba7269c15cf7
# good: [f1eeed7a29fef790f4ff53fb972779a3b3634c46] ised: 2.6.0 -> 2.7.0
git bisect good f1eeed7a29fef790f4ff53fb972779a3b3634c46
# bad: [8efa3745c7b4101b6981682d08e597b55ba0b6d5] libmwaw: 0.3.5 -> 0.3.7
git bisect bad 8efa3745c7b4101b6981682d08e597b55ba0b6d5
# good: [1406e8e6936af1bde991e93289d88ca478f38bc2] angelscript: 2.30.1 -> 2.30.2
git bisect good 1406e8e6936af1bde991e93289d88ca478f38bc2
# bad: [c8b975388235823322a0962f88f05594229435e3] eigen: 3.2.5 -> 3.3-alpha1
git bisect bad c8b975388235823322a0962f88f05594229435e3
# good: [1dba4b4e6af2c427d2e21986e253fff45df485ec] scons: 2.3.6 -> 2.4.1
git bisect good 1dba4b4e6af2c427d2e21986e253fff45df485ec
# good: [8d81c0ada284da64cc6d31bbd48957c9bf659ec9] cdk: a new release…
git bisect good 8d81c0ada284da64cc6d31bbd48957c9bf659ec9
# bad: [b2ac241e958c767c4d817e65c37802014499d7a4] iptables: 1.4.21 -> 1.6.0
git bisect bad b2ac241e958c767c4d817e65c37802014499d7a4
# first bad commit: [b2ac241e958c767c4d817e65c37802014499d7a4] iptables: 1.4.21 -> 1.6.0
```
And indeed, reverting b2ac241e958c767c4d817e65c37802014499d7a4 on top of 2ee6396 produced a working system.

**Comment by ktosiek** [comment]
Looks like it won't be fixed until next connman (1.32 probably?) as it was only commited 3 days ago: http://git.kernel.org/cgit/network/connman/connman.git/commit/?id=acea08a0e4234a4c1a87bedc087c73ff36de0c7b

**Comment by zimbatm** [comment]
Looks like this is resolved now
